src/minidom_utils/mod.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+//! Some utilities to use Minidom and returns [Result](crate::Result) when
+//! parsing instead of [Option](Option)
+
+mod try_attribute;
+pub use try_attribute::TryAttribute;
+mod try_only_child;
+pub use try_only_child::TryOnlyChild;
```

Comment:
why not put everything here?

Reply:
Small modules, easy to maintain?  I took a look on the web and here is some quotes from a [Reddit thread](https://www.reddit.com/r/rust/comments/alsph9/rusts_modules_and_project_organization_best/) that I found useful.

> For example, a suggestion I found once was: one module per trait/type definition.

> A useful not Rust-specific idea is that _users_ of a module prefer a flat namespace, while _implementers_ prefer hierarchical namespace. So nested module structure plus a flat facade with re-exports as a crate's interface is usually a good idea